documentation/modules/exploit/linux/local/ufo_privilege_escalation.md

```diff
@@ -0,0 +1,190 @@
+## Vulnerable Application
+
+This module attempts to gain root privileges on Linux systems by abusing UDP Fragmentation Offload (UFO).
+
+This module has been tested successfully on:
+
+  * Ubuntu 14.04.5 4.4.0-31-generic x64 Desktop
+  * Ubuntu 16.04 4.8.0-53-generic
+  * Linux Mint 18 4.8.0-58-generic
+  
+There is a large swath of vulnerable kernels believed to be from about 3.13 to 4.11.
+Only Ubuntu based kernels were coded into this particular exploit, however other flavors of linux
+may also be vulnerable.  The following kernels and systems are coded into the exploit and should work,
+but may not be tested.
```

Comment:
It may be worth mentioning that failed exploitation attempts will cause the system to freeze, requiring reboot.

Reply:
In my testing this has never happened. The exploit is stable and the ROP chain is reliable.

However, while adding new targets, I froze the system a bunch of times.

As an aside, I ran into issues adding xenial `4.8.0-42-generic` and `4.8.0-44-generic` kernels. The ROP chain does not work and freezes the system. I never bothered to investigate why. These targets are **not** in the original C code, nor the modified C code, and as such, the exploit will never attempt to exploit these kernels.